[DOCS] NPU properties in documentation for master (#25600)
openvinotoolkit/openvino: docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/npu-device.rst

   based on the performance mode is **4 for THROUGHPUT** and **1 for LATENCY**.
   The default mode for the NPU device is LATENCY.

**ov::intel_npu::compilation_mode_params**

> ``ov::intel_npu::compilation_mode_params`` is an NPU-specific property that allows to
> control model compilation for NPU.

Reviewer: ```suggestion
``ov::intel_npu::compilation_mode_params`` is an NPU-specific property that allows
control of model compilation for NPU.
```

.. note::

>    The functionality is in experimental stage currently, can be a subject for
>    deprecation and may be replaced with generic OV API in future OV releases.

Reviewer: ```suggestion
   The functionality is in an experimental stage currently, can be a subject for
   deprecation and may be replaced with a generic OV API in future OV releases.
```

> Following configuration options are supported:

Reviewer: ```suggestion
The following configuration options are supported:
```

**optimization-level**

Defines a preset of optimization passes to be applied during compilation.

.. list-table::
   :widths: 10 200
   :header-rows: 1

   * - **Value**
     - **Description**
   * - 0

>      - Reduced subset of optimization passes. Smaller compile time.

Reviewer: ```suggestion
     - A reduced subset of optimization passes. Smaller compile time.
```

   * - 1
     - **Default.** Balanced performance/compile time.
   * - 2
     - Prioritize performance over compile time that may be an issue.

**performance-hint-override**

> An extension for LATENCY mode being specified using ``ov::hint::performance_mode``

Reviewer: ```suggestion
An extension for LATENCY mode being specified using ``ov::hint::performance_mode``.
```

> Has no effect for other ``ov::hint::PerformanceMode`` hints.

Reviewer: ```suggestion
It has no effect on the other ``ov::hint::PerformanceMode`` hints.
```
